src/textual/widgets/_rule.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+
+from rich.text import Text
+from typing_extensions import Literal
+
+from .._border import BORDER_CHARS
+from ..app import RenderResult
+from ..css._error_tools import friendly_list
+from ..css.constants import VALID_BORDER
+from ..css.types import EdgeType
+from ..reactive import Reactive, reactive
+from ..widget import Widget
+
+RuleOrientation = Literal["horizontal", "vertical"]
+
+LineStyle = EdgeType
+
+
+_VALID_RULE_ORIENTATIONS = {"horizontal", "vertical"}
+"""The valid orientations of the rule widget"""
+
+_VALID_LINE_STYLES = VALID_BORDER
+"""The valid line styles of the rule widget"""
+
+
+class InvalidRuleOrientation(Exception):
+    """Exception raised for an invalid rule orientation."""
+
+
+class InvalidLineStyle(Exception):
+    """Exception raised for an invalid rule line style."""
+
+
+class Rule(Widget, can_focus=False):
+    """A rule widget to separate content, similar to a <hr> HTML tag."""
+
+    DEFAULT_CSS = """
+    Rule {
+        color: $primary;
+    }
+
+    Rule.-horizontal {
+        max-height: 1;
+        margin: 1 0;
+    }
+
+    Rule.-vertical {
+        max-width: 1;
+        margin: 0 2;
+    }
+    """
+
+    orientation: Reactive[RuleOrientation] = reactive[RuleOrientation]("horizontal")
+    """The orientation of the rule."""
+
+    line_style: Reactive[LineStyle] = reactive[LineStyle]("solid")
+    """The line style of the rule."""
+
+    def __init__(
+        self,
+        orientation: RuleOrientation = "horizontal",
+        line_style: LineStyle = "solid",
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ) -> None:
+        """Initialize a rule widget.
+
+        Args:
+            orientation: The orientation of the rule.
+            line_style: The line style of the rule.
+            name: The name of the widget.
+            id: The ID of the widget in the DOM.
+            classes: The CSS classes of the widget.
+            disabled: Whether the widget is disabled or not.
+        """
+        super().__init__(name=name, id=id, classes=classes, disabled=disabled)
+        self.orientation = self.validate_orientation(orientation)
+        self.line_style = self.validate_line_style(line_style)
```

Comment:
I believe you don't need to call the validators explicitly here.

(Come to think of it, this may be done elsewhere in some other widgets, but I think that's just a vestige of some old reactive attributes limitation.)

Reply:
It _shouldn't_ be necessary.